maps/game_map.go

```diff
@@ -44,6 +44,20 @@ func (d sizes) IsUnlimited() bool {
 	return len(d) == 1 && d[0].Width == 0
 }
 
+func (d sizes) IsAllowable(Width uint, Height uint) bool {
+	if d.IsUnlimited() {
+		return true
+	}
+
+	for _, size := range d {
+		if size.Width == Width && size.Height == Height {
```

Comment:
Might be a little nicer to cast to `uint` internally here so that most calls to `IsAllowable` don't need to cast int -> uint.